api/src/vaccination-records/vaccination-records.service.spec.ts

```diff
@@ -16,4 +19,76 @@ describe('VaccinationRecordsService', () => {
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  it('should return all records', async () => {
+    repositoryMocks.vaccinationRecords.find.mockResolvedValue(
+      vaccinationRecords,
+    );
+
+    const record = await service.findAll();
+
+    expect(record).toEqual(vaccinationRecords);
+  });
+
+  it('should return a record by id', async () => {
+    repositoryMocks.vaccinationRecords.findOne.mockResolvedValue(
+      vaccinationRecords[0],
+    );
+
+    const record = await service.findOne(1);
+
+    expect(repositoryMocks.vaccinationRecords.findOne).toHaveBeenCalledWith({
+      relations: ['vaccine', 'patient'],
+      where: { id: 1 },
+    });
+    expect(record).toEqual(vaccinationRecords[0]);
+  });
+
+  it('should create a vaccination record', async () => {
+    const recordToCreate: CreateVaccinationRecordDto = {
+      patientId: 1,
+      vaccineId: 1,
+      applierName: 'John Doe',
+    };
+
+    repositoryMocks.vaccinationRecords.save.mockResolvedValue(recordToCreate);
+
+    const record = await service.create(recordToCreate);
+
+    expect(repositoryMocks.vaccinationRecords.save).toHaveBeenCalledWith({
+      patient: { id: recordToCreate.patientId },
+      vaccine: { id: recordToCreate.vaccineId },
+      applierName: recordToCreate.applierName,
+    });
+
+    expect(record).toEqual(recordToCreate);
+  });
+
+  it('should update a record', async () => {
+    const recordToUpdate: UpdateVaccinationRecordDto = {
+      patientId: 1,
+      vaccineId: 1,
+      applierName: 'John Doe',
+    };
+
+    const expectedRecord = { ...recordToUpdate, id: 1 };
+
+    repositoryMocks.vaccinationRecords.save.mockResolvedValue(expectedRecord);
+
+    const record = await service.update(expectedRecord.id, recordToUpdate);
+
+    expect(repositoryMocks.vaccinationRecords.save).toHaveBeenCalledWith({
+      patientId: recordToUpdate.patientId,
+      vaccineId: recordToUpdate.vaccineId,
+      applierName: recordToUpdate.applierName,
+      id: expectedRecord.id,
+    });
+
+    expect(record).toEqual(expectedRecord);
+  });
```

Comment:
Como temos stubs pra patients e pra vaccines, devemos evitar ao máximo usar "números mágicos" chumbados no código, como é o caso do "1" aqui em `patientId` e `vaccineId`. 
Devemos importar os stubs e usar os ids deles. 
Por exemplo:

```ts
    const recordToCreate: CreateVaccinationRecordDto = {
      patientId: patientItems[0].id,
      vaccineId: vaccineItems[0].id,
      applierName: 'John Doe',
    };

```

O mock não deve retornar `expectedRecord` porque `expectedRecord` é to tipo `CreateVaccinationRecordDto`, que é o tipo que nosso serviço espera receber. 
O mock deve retornar a entidade do banco de dados, ou seja `VaccinationRecord`.

Reply:
Entendi caro colega de equipe, farei o possível para corrigir o quanto antes!